server/controllers/complaintHandler.js

```diff
@@ -115,4 +151,66 @@ const deleteUserComplaint = (req, res) => {
   }
 }
 
-module.exports = {createDocComplaint, retrieveDocComplaint, getDocComplaintList, deleteDocComplaint, createUserComplaint, retrieveUserComplaint, getUserComplaintList, deleteUserComplaint};
+const retrieveUnprocessedUserComplaints = (req, res) => {
+  UserComplaint.find({processed: false}, function(err, results){
+    if (err || !result) {
+      res.status(500).json({error:"Could not retrieve User Complaints."});
+    }
+    res.status(200).json({unprocessedUserComplaints:results});
+  });
+}
+
+const processUserComplaint = (req, res) => {
+  if(!req.isAuthenticated()){
+    res.redirect('/');
+  }
+  else {
+    UserComplaint.findById(req.body.userComplaintId, function (err, result) {
+      if(err || !result){
+        res.status(500).json({error:"Could not retrieve the User Complaint for updating."});
+      }
+      else {
+        result.processed = true;
+        result.save(function (saveerr, updatedresult) {
+          if(saveerr || !updatedresult){
+            res.status(500).json({error:"Could not update the User Complaint."});           
+          }
+          else {
+            console.log("User Complaint in UserComplaints collection updated.");
+            res.status(200).json({msg:"User Complaint Updated."});
+          }
+        });
+      }
+    });
+  }
+}
+
+const getCurrentUserComplaints = (req, res) => {
+  if(!req.isAuthenticated()){
+    res.redirect('/');
+  }
+  else {
+    UserComplaint.find({targetUserId:req.user._id}, function(err, results) {
+      if (err || !result) {
+        res.status(500).json({error:"Could not retrieve current User Complaints."});
+      }
+      res.status(200).json({currentUserComplaintList:results});
+    });
+  }
+}
+
+const getCurrentUserComplaints = (req, res) => {
+  if(!req.isAuthenticated()){
+    res.redirect('/');
+  }
+  else {
+    UserComplaint.find({fromUserId:req.user._id}, function(err, results) {
```

Comment:
this should also check if the request wants to see processed or unprocessed complaints, i.e. in req.body.getProccessed

Reply:
ok